### PR TITLE
Package split

### PR DESCRIPTION
**Pull request by sapphire-arches**
Splits the crate in two, one for the frontend and one for the core. This is still WIP, and probably breaks the CI config.

### CODE REVIEW

**Comment by yupferris** [comment]
Thanks for this :) a couple things:

- I believe the CI build failures happened because I accepted a PR that made a small change in the VSU before you moved it. Rebasing on master will likely fix this (and since I don't think you changed the VSU, if there's a conflict it shouldn't be too bad).
- Was there a reason to bump all the crate versions? I'm guessing this happened because you deleted `Cargo.lock` for the cli or didn't move it when splitting the crates or something. This is likely fine; in fact, even as a side-effect it looks like we'll get https://github.com/tomaka/cpal/pull/146 for free, but it would have been nice to try and preserve those through a change like this and update the dependencies separately so we can go through the possibly new dependencies and any licensing side-effects we might incur. Not a huge deal though.

Apart from that I haven't dug very deep here, but unless you'd rather I do so I'd kindof just like to let you do your thing freely until you need any assistance/reviewing if that works for you :)

**Comment by sapphire-arches** [comment]
I merged the VSU change and I think I've fixed the appveyor config issue. If you're especially opposed to merge commits I can rewrite history to remove it.

**Comment by yupferris** [comment]
Normally I'm not opposed to those but in this case I don't want to have the repo in any intermediate states (or any commits specifically) where a build described by the CI could produce something that doesn't line up license-wise, so if it wouldn't be too much trouble I think it'd be best to rewrite the history this time :)

**Comment by sapphire-arches** [comment]
I squashed all of my commits in to one that (should) pass CI. This should be ready for review, it mostly just consists of copying some files to the right places and tweaking imports. There's definitely a lot more interesting work that can be done, including making the CLI stuff a bit more robust and documenting/cleaning up the public interface to the emulator core but that's probably best left to another PR.

File: rustual-boy-core/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+extern crate encoding;
+
+#[macro_use]
+extern crate nom;
```
Comment (yupferris) [inline rustual-boy-core/src/lib.rs]: I don't think `nom` or `futures` are required from the core; I think only the frontend ends up using them (`futures` for some glue for CPAL and `nom` for command parsing iirc). If this is the case we should not require them here.

**Review by yupferris — CHANGES_REQUESTED** [review]
I think the only thing is extra crate deps for the core that shouldn't be necessary. If those can be removed we should be good to go :)

**Comment by sapphire-arches** [comment]
Should be fixed, I just removed the dependencies and it still built locally so it should be fine on CI.

**Review by yupferris — APPROVED** [review]

**Comment by yupferris** [comment]
lgtm, thanks :)

**Pull request merged**